GSA/made-in-america
# [MIA-248] bug/ fix pagination render

closes https://cm-jira.usa.gov/browse/MIA-248

## Description

includes `ngDoCheck` callback method that performs our needed change-detection

## How to test

- visit [waivers page](https://federalist-c275c827-2779-42ff-ac19-0331ebcacca9.app.cloud.gov/preview/gsa/made-in-america/bug/gsq-fix-pagination-render/waivers/)
- select "Reviewed"
- click through pagination
- select "All"
- click through pagination
- select "Reviewed" again
- click through pagination

paginator should be behaving as expected without a gap in the index of the rendered buttons, or duplication buttons

To prepare for code review, please follow the following checks.
